Library/Homebrew/test/utils/svn_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "utils/svn"
+
+describe Utils do
+  describe "#self.svn_available?" do
+    before(:each) do
+      if described_class.instance_variable_defined?(:@svn)
+        described_class.send(:remove_instance_variable, :@svn)
+      end
+    end
+
+    it "returns svn version if svn available" do
+      expect(described_class.svn_available?).to be_truthy
+    end
+  end
+
+  describe "#self.svn_remote_exists" do
+    it "returns true when svn is not available" do
+      allow(Utils).to receive(:svn_available?).and_return(false)
+      expect(described_class.svn_remote_exists("blah")).to be_truthy
+    end
+
+    context "when svn is available" do
+      before do
+        allow(Utils).to receive(:svn_available?).and_return(true)
+        url.mkpath
+      end
+
+      let(:url) { mktmpdir/"blah" }
+
+      it "returns false when remote does not exist" do
+        expect(described_class.svn_remote_exists(url)).to be_falsey
+      end
+
+      it "returns true when remote exists", :needs_network do
+        remote = "http://amethyst.yui.ne.jp/svn/pc98/np2/trunk/"
```

Comment:
Consider using https://github.com/Homebrew/install as it's really small and we control it (you can clone GitHub repositories as Subversion repositories).